library/agent/AIStatistics.ts

```diff
@@ -36,20 +55,54 @@ export class AIStatistics {
           output: 0,
           total: 0,
         },
+        routesLRU: new LRUMap<string, AIRouteStats>(this.maxRoutes),
       });
     }
 
     return this.calls.get(key)!;
   }
 
+  private ensureRouteStats(
```

Comment:
If we use the existing Routes object would also remove need for this (and the LRU)